.mypy.ini

```diff
@@ -1,5 +1,5 @@
 [mypy]
-python_version = 3.8
+python_version = 3.9
```

Comment:
This would have to go into a separate PR and be coupled with changes to the core packaging metadata, linting configuration, dev env and CI setup.